src/net/l3/arp.c

```diff
@@ -156,32 +150,30 @@ static int arp_hnd_request(const struct arphdr *arph,
 
 	/* declone sk_buff */
 	if (NULL == skb_declone(skb)) {
-		DBG(printk("arp_hnd_request: no memory\n"));
+		log_error("arp_hnd_request: no memory\n");
 		skb_free(skb);
 		return -ENOMEM;
 	}
 
-	DBG({
-		printk("arp_hnd_request: send reply with ");
-		if (arph->ar_pro == ntohs(ETH_P_IP)) {
-			struct in_addr in;
-			assert(arph->ar_pln == sizeof in);
-			memcpy(&in, dst_paddr, sizeof in);
-			printk("%s", inet_ntoa(in));
-		}
-		else {
-			printk("unknown(%x)", htons(arph->ar_pro));
-		}
-		if (arph->ar_hrd == ntohs(ARP_HRD_ETHERNET)) {
-			assert(arph->ar_hln == ETH_ALEN);
-			printk("[" MACADDR_FMT "]",
-				MACADDR_FMT_ARG(dst_haddr));
-		}
-		else {
-			printk("[unknown(%x)]", htons(arph->ar_hrd));
-		}
-		printk("\n");
-	});
+	log_debug("arp_hnd_request: send reply with ");
```

Comment:
Can you extract blocks like that in a routine?
You should use OPTION_GET(NUMBER,log_level) macro